framework/src/org/checkerframework/common/value/ValueAnnotatedTypeFactory.java

```diff
@@ -171,15 +182,41 @@ public Void visitDeclared(AnnotatedDeclaredType type, Void p) {
          * If any constant-value annotation has &gt; MAX_VALUES number of values provided, treats
          * the value as UnknownVal. Works together with ValueVisitor.visitAnnotation, which issues a
          * warning to the user in this case.
+         *
+         * <p>Additional rules applied for IntVal and IntRange annotations: if anno is IntVal and
```

Comment:
@wmdietl You are right about modifying the ValueVisitor.java accordingly, which I forgot. The question is, what kind of warning info should we give users?

My current thought is:
1. if (@IntRange with too narrow range) too.narrow.range.given, replace with @IntVal
2. if (@IntVal with too many values) too.many.int.values, replace with @IntRange
3. if (other annos with too many values) too.many.values.given, replace with @UnknownVal